internal/agent/device/resource/resource.go

```diff
@@ -0,0 +1,202 @@
+package resource
+
+import (
+	"context"
+	"math"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/flightctl/flightctl/pkg/log"
+)
+
+type Manager interface {
+	Usage() (*Usage, error)
+	Run(ctx context.Context)
+}
+
+type Collector[T any] interface {
+	Usage() *T
+	Run(ctx context.Context)
+}
+
+type ResourceManager struct {
+	filesystem Collector[FsUsage]
+	log        *log.PrefixLogger
+}
+
+// NewManager creates a new resource Manager.
+func NewManager(
+	log *log.PrefixLogger,
+	fsAlertThreshold int,
+	fsWarnThreshold int,
+	fsPaths []string,
+	fsSyncDuration time.Duration,
+	fsSyncTimeout time.Duration,
+) Manager {
+	return &ResourceManager{
+		filesystem: NewFs(
+			log,
+			fsAlertThreshold,
+			fsWarnThreshold,
+			fsSyncDuration,
+			fsSyncTimeout,
+			fsPaths,
+		),
+		log: log,
+	}
+}
+
+func (m *ResourceManager) Run(ctx context.Context) {
+	m.log.Infof("Starting resource manager...")
+	defer m.log.Infof("Resource manager stopped")
+
+	go m.filesystem.Run(ctx)
+
+	<-ctx.Done()
+}
+
+func (m *ResourceManager) Usage() (*Usage, error) {
+	return &Usage{
+		FsUsage: m.filesystem.Usage(),
+	}, nil
+}
+
+type Usage struct {
+	FsUsage *FsUsage
+}
+
+type FsUsage struct {
+	alertThreshold int
+	warnThreshold  int
+	Inodes         uint64
+	Total          uint64
+	Free           uint64
+	Used           uint64
+}
+
+func (u *FsUsage) IsAlert() bool {
+	avail := percentageAvailable(u.Free, u.Total)
+	return avail < u.alertThreshold
+}
+
+func (u *FsUsage) IsWarn() bool {
+	avail := percentageAvailable(u.Free, u.Total)
+	return avail < u.warnThreshold
+}
+
+var _ Collector[FsUsage] = (*Fs)(nil)
+
+type Fs struct {
+	mu           sync.Mutex
+	syncDuration time.Duration
+	syncTimeout  time.Duration
+	fsPaths      []string
+	usage        FsUsage
+	log          *log.PrefixLogger
+}
+
+func NewFs(
+	log *log.PrefixLogger,
+	alertPercentThreshold int,
+	warnPercentThreshold int,
+	syncDuration,
+	syncTimeout time.Duration,
+	paths []string,
+) *Fs {
+	return &Fs{
+		syncDuration: syncDuration,
+		syncTimeout:  syncTimeout,
+		fsPaths:      paths,
+		usage: FsUsage{
+			alertThreshold: alertPercentThreshold,
+			warnThreshold:  warnPercentThreshold,
+		},
+		log: log,
+	}
+}
+
+func (f *Fs) Run(ctx context.Context) {
+	ticker := time.NewTicker(f.syncDuration)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if err := f.sync(ctx); err != nil {
+				f.log.Errorf("Error syncing filesystem usage: %v", err)
```

Comment:
Is there any way to get this information to the user?  For example, if they misconfigured the monitoring?